### PR TITLE
Fix for issue #83

### PR DESCRIPTION
**Pull request by Chopknee**
Changed raw key access to a get with safe default when dealing with IMDB returned data.

### CODE REVIEW

File: movienightbot/db/controllers.py [hunk 1]
```diff
@@ -102,13 +102,14 @@ def update_imdb_id(self, server_id: int, movie_name: str, imdb_id: str):
...
             return 0
+
         imdb_data = {
             "imdb_id": imdb_info.movieID,
             "title": imdb_info["title"],
-            "canonical_title": imdb_info["canonical title"],
-            "year": imdb_info["year"],
-            "thumbnail_poster_url": imdb_info["cover url"],
-            "full_size_poster_url": imdb_info["full-size cover url"],
+            "canonical_title": imdb_info.get("canonical title", "N/A"),
```
Comment (squirrelo) [inline movienightbot/db/controllers.py]: this should probably default to `imdb_info["title"]` for ease of use

File: setup.py [hunk 2]
```diff
@@ -3,11 +3,12 @@
 from movienightbot import __version__ as bot_version
 
 requirements = [
-    "py-cord[speed]>=2.0",
+    "py-cord[speed]==2.4.0",
```
Comment (squirrelo) [inline setup.py]: why the pin and not `>=2.4`?

File: movienightbot/commands/suggest.py [hunk 3]
```diff
@@ -40,10 +40,10 @@ def imdb_data(movie: str, kind: str) -> Tuple[Union[None, IMDBInfo], Union[None,
     imdb_row_data = {
         "imdb_id": imdb_info.movieID,
         "title": imdb_info["title"],
-        "canonical_title": imdb_info["canonical title"],
-        "year": imdb_info["year"],
-        "thumbnail_poster_url": imdb_info["cover url"],
-        "full_size_poster_url": imdb_info["full-size cover url"],
+        "canonical_title": imdb_info.get("canonical title", "N/A"),
```
Comment (squirrelo) [inline movienightbot/commands/suggest.py]: same here

File: setup.py [hunk 4]
```diff
@@ -8,6 +8,7 @@
     "marshmallow",
     "pyyaml",
     "cinemagoer>=2022.12.27",
+    "discord.py>=2.3.2"
```
Comment (Chopknee) [inline setup.py]: If this package isn't installed, the bot errors out and does not start. It is not installed as a dependency of other packages.

**Review by Chopknee — COMMENTED** [review]
Addressed the issues brought up.

Sorry, I'm mostly unfamiliar with using github. :)

File: movienightbot/db/controllers.py [hunk 5]
```diff
@@ -102,13 +102,14 @@ def update_imdb_id(self, server_id: int, movie_name: str, imdb_id: str):
...
         imdb_data = {
             "imdb_id": imdb_info.movieID,
             "title": imdb_info["title"],
-            "canonical_title": imdb_info["canonical title"],
-            "year": imdb_info["year"],
-            "thumbnail_poster_url": imdb_info["cover url"],
-            "full_size_poster_url": imdb_info["full-size cover url"],
+            "canonical_title": imdb_info.get("canonical title", imdb_info["title"]),
+            "year": imdb_info.get("year", 0),
+            "thumbnail_poster_url": imdb_info.get("cover url", imdb_info.get("full-size cover url", "")),
```
Comment (squirrelo) [inline movienightbot/db/controllers.py]: This flipped full size into thumbnail and vice versa

File: movienightbot/commands/suggest.py (hunk 3 above)
Comment (squirrelo) [inline movienightbot/commands/suggest.py]: This still needs to be title for default canonical title value

**Review by squirrelo — APPROVED** [review]

**Pull request merged**